### DISCUSSION TITLE
PASOE Debug Adapter

### DISCUSSION THREAD

**Discussion by gquerret** [vscode-abl/vscode-abl#134] [General]
This discussion is for PASOE Debug Adapter related questions and comments.

**Comment by M0rr155** [comment]
With my build path to the same dir as localRoot setting it works !!
![image](https://github.com/vscode-abl/vscode-abl/assets/164477200/67e63114-adae-44d2-a703-934419d1a208)

now my .r are in the same location as my source code

but what when i want my .r files in a different dir?

**Reply by gquerret** [reply]
The debugger session is tied to the project which is opened in VSCode, so it inherits the project configuration. It is not meant to be used to connect to any PASOE instance, and expect breakpoints to work automatically. Breakpoints can only be set if rcode can be read because the line numbers mapping is only available here. And it of course needs source code to step into the code.
Note that a current limitation is that the debugger session can only be tied to a single project, even if VSCode has multiple projects in the workspace.

**Comment by M0rr155** [comment]
Thank you Gilles,

Can you tell me about  ${workspaceFolder}"
Which folder is this?


  1.
Is this the folder where openedge-project.json is in (C:\VSCodeW\backend\)?
OR
  2.
Is this the folder of the VSCode workspace (C:\VSCodeW) ?

**Comment by gquerret** [comment]
https://code.visualstudio.com/docs/editor/variables-reference
